src/data-crawler/infrastructure/gateways/établissement-territorial-loader/FinessXmlÉtablissementTerritorialSourceExterneLoader.ts

```diff
@@ -230,17 +232,28 @@ export class FinessXmlÉtablissementTerritorialLoader implements ÉtablissementT
 
     const catégories = this.convertXmlToJs.handle<CatégorieFluxFiness>(cheminDuFichierCatégorie)
 
+    const numéroFinessDesEntitésJuridiques = await this.entitéJuridiqueHeliosLoader.récupèreLeNuméroFinessDesEntitésJuridiques()
+
     const établissementTerritorialFluxFinessIdentité = this.convertXmlToJs.handle
       <ÉtablissementTerritorialIdentitéFluxFiness>(cheminDuFichierÉtablissementTerritorialIdentité)
 
-    const établissementsTerritoriauxIdentité = établissementTerritorialFluxFinessIdentité.fluxfiness.structureet.map(
-      (établissementTerritorialIdentité: ÉtablissementTerritorialIdentitéFiness) =>
-        this.construitÉtablissementTerritorialIdentité(établissementTerritorialIdentité, dateDeMiseAJourDeLaSource, catégories)
-    )
+    const établissementsTerritoriauxIdentité = établissementTerritorialFluxFinessIdentité.fluxfiness.structureet
+      .filter(this.gardeLesÉtablissementsOuverts(numéroFinessDesEntitésJuridiques))
+      .map((établissementTerritorialIdentitéFiness: ÉtablissementTerritorialIdentitéFiness) =>
+        this.construitÉtablissementTerritorialIdentité(établissementTerritorialIdentitéFiness, dateDeMiseAJourDeLaSource, catégories))
 
     return établissementsTerritoriauxIdentité
   }
 
+  private gardeLesÉtablissementsOuverts(numéroFinessDesEntitésJuridiques: string[]) {
+    return (établissementTerritorialIdentitéFiness: ÉtablissementTerritorialIdentitéFiness) => {
+      const établissementTerritorialCaduque = établissementTerritorialIdentitéFiness.indcaduc._text === 'O'
+      const établissementTerritorialFermé = établissementTerritorialIdentitéFiness.datefermeture._text !== undefined
+      const établissementJuridiqueOuvert = numéroFinessDesEntitésJuridiques.includes(établissementTerritorialIdentitéFiness.nofinessej._text as string)
```

Comment:
Tu peux aller jusqu'à isoler chaque booléen calculé ici dans une méthode :
```suggestion
      const établissementTerritorialCaduque = this.établissementEstCaduque(établissementTerritorialIdentitéFiness)
      const établissementTerritorialFermé = this.établissementEstFermé(établissementTerritorialIdentitéFiness)
      const entitéJuridiqueOuvert = this.entitéJuridiqueDeRattachementEstOuverte(établissementTerritorialIdentitéFiness )```